### PR TITLE
Unify spin.toml in Spin key-vaule docs

### PR DESCRIPTION
**Pull request by ThorstenHans**
`spin.toml` had quite some differences across Rust, TypeScript, and TinyGo. With this commit, all manifests rely on the same naming-pattern, provide a description, and have inline objects formatted the same way

Content must go through a pre-merge checklist.

## Note on `bart check`

`bart check` ran without errors / warnings for the file I touched. However, there are some findings in other files located in `/content/spin`

## Pre-Merge Content Checklist

This documentation has been checked to ensure that:

- [x] File does not use CRLF, but uses plain LF (hint: use `cat -ve <filename> | grep '^M' | wc -l` and expect 0 as a result) 
- [x] Has passed [`bart check`](https://developer.fermyon.com/bartholomew/quickstart)
- [x] Has run `npm run build-index`
- [x] Has been manually tested by running in Spin/Bartholomew (hint: use `PREVIEW_MODE=1` and run `npm run styles` to update styling)
- [x] Code blocks have the programming language set to properly highlight syntax and the proper copy directive
- [x] Have tested with `npm run test` and resolved all errors

### CODE REVIEW

File: content/spin/kv-store.md [hunk 1]
```diff
@@ -65,7 +65,7 @@ As previously documented, you can go ahead and [create a new Spin application fr
 <!-- @selectiveCpy -->
 
 ```bash
-$ spin new http-rust spin-key-value
+$ spin new http-rust spin-rust-key-value
```
Comment (itowlson) [inline content/spin/kv-store.md:68]: Given that all these blocks are under per-language tabs, I'm not sure if there's any value in having different names.  Would it be bad if we called them all plain `spin-key-value`?  I don't have strong views, more raising the question for @tpmccallum.

**Review by itowlson — COMMENTED** [review]
Thanks for this!  I had one thought but probably as much for us as for you...!

**Comment by itowlson** [comment]
I think the failing action may be a link already in `main` that is "in waiting" but not live yet.

**Comment by ThorstenHans** [comment]
> Thanks for this!  I had one thought but probably as much for us as for you...!

Thought about that too. Will update the PR once back at the mac

**Review by itowlson — APPROVED** [review]
LGTM - thanks!

File: content/spin/kv-store.md (hunk 1 above)
Comment (tpmccallum) [reply content/spin/kv-store.md]: Consistency is welcome across all tabs; greatly appreciated and about to be merged ...

**Comment by tpmccallum** [comment]
Thanks again @ThorstenHans 
Greatly appreciated!

**Pull request merged**